### ISSUE TITLE
`no-return-array-push` false positive on next.js router

### ISSUE THREAD

**Issue by Sparticuz** [sindresorhus/eslint-plugin-unicorn#3445]
Similar to #3431, `no-return-array-push` is getting a false positive on a `next/navigation` router because it uses the function `.push()`. This rule should be restricted to arrays, not just any object with a `.push()` method

`no-return-array-push`

```js
router.push(`/${region}/customers/${String(result.data)}` as Route)
```
